cf_xarray/accessor.py

```diff
@@ -1609,12 +1609,16 @@ def add_bounds(self, dims: Union[Hashable, Iterable[Hashable]]):
         The bounds variables are automatically named f"{dim}_bounds" where ``dim``
         is a dimension name.
         """
-        if isinstance(dims, Hashable):
-            dimensions = (dims,)
-        else:
-            dimensions = dims
+        if isinstance(keys, str):
+            keys = [keys]
+
+        dimensions = set()
+        for key in keys:
+            dimensions |= set(
+                apply_mapper(_get_dims, self._obj, key, error=False, default=[key])
+            )
```

Comment:
A set comprehension is cleaner here


```suggestion
        dimensions = {
                apply_mapper(_get_dims, self._obj, key, error=False, default=[key])
                for key in keys
            )
```

Reply:
Actually shouldn't the default be `key` instead of `[key]` given that you're doing set subttraction later.

Reply:
`apply_mapper` returns a list of dimension names, shouldn't `default` do the same?

Regarding the set comprehension, this wouldn't work: `{["dim1", "dim2"], ["dim3"]}`.

How about this?
```python
dimensions = {dim for key in keys for dim in apply_mapper(_get_dims, self._obj, key, error=False, default=[key])}
```

Or we could use update:
```python
dimensions = set()
for key in keys:
    dimensions.update(apply_mapper(_get_dims, self._obj, key, error=False, default=[key]))
```

Reply:
I like update